.github/workflows/build-and-test.yml

```diff
@@ -25,69 +25,23 @@ jobs:
   build:
     if: inputs.build_script != '' && inputs.build_image != ''
     name: Build ${{inputs.compiler}}${{inputs.compiler_version}}/C++${{inputs.std}}
-    runs-on: linux-${{inputs.cpu}}-cpu16
-    container: 
-      options: -u root
-      image: ${{ inputs.build_image }}
-    permissions:
-      id-token: write
-    steps:
-      - name: Checkout repo
-        uses: actions/checkout@v3
-        with:
-          path: cccl
-      # In order for sccache to be shared between CI and lcaal devcontainers, code needs to be in the same absolute path
-      # This ensures the code is in the same path as it is within the devcontainer
-      - name: Move files to coder user home directory
-        run: |
-          cp -R cccl /home/coder/cccl
-          chown -R coder:coder /home/coder/
-      - name: Configure credentials and environment variables for sccache
-        uses: ./cccl/.github/actions/configure_cccl_sccache
-      - name: Run build script
-        shell: su coder {0}
-        run: | 
-            cd ~/cccl
-            cmd="${{ inputs.build_script }} \"${{inputs.compiler_exe}}\" \"${{inputs.std}}\" \"${{inputs.gpu_build_archs}}\""
-            eval $cmd || exit_code=$?
-            if [ ! -z "$exit_code" ]; then
-                echo "::error::Build failed! To checkout the corresponding code and reproduce this build locally, run the following commands:" 
-                echo "git clone --branch $GITHUB_REF_NAME --single-branch --recurse-submodules https://github.com/$GITHUB_REPOSITORY.git && cd $(echo $GITHUB_REPOSITORY | cut -d'/' -f2) && git checkout $GITHUB_SHA"
-                echo "docker run --rm -it --gpus all --pull=always --volume \$PWD:/repo --workdir /repo ${{ inputs.build_image }} $cmd"
-                echo "Alternatively, for a more convenient, interactive environment to reproduce the issue, you can launch a devcontainer in vscode:"
-                echo "git clone --branch $GITHUB_REF_NAME --single-branch --recurse-submodules https://github.com/$GITHUB_REPOSITORY.git && cd $(echo $GITHUB_REPOSITORY | cut -d'/' -f2) && git checkout $GITHUB_SHA"
-                echo ".devcontainer/launch.sh ${{inputs.cuda_version}} ${{inputs.compiler}}${{inputs.compiler_version}}"
-                echo "Then, open a terminal inside vscode (ctrl+shift+\`) and run:"
-                echo "$cmd"
-                exit $exit_code
-            fi
+    uses: ./.github/workflows/run-as-coder.yml
+    with:
+      name: Build ${{inputs.compiler}}${{inputs.compiler_version}}/C++${{inputs.std}}
+      runner: linux-${{inputs.cpu}}-cpu16
+      image: ${{inputs.build_image}}
+      command: |
+        ${{ inputs.build_script }} "${{inputs.compiler_exe}}" "${{inputs.std}}" "${{inputs.gpu_build_archs}}"
+
   test:
     needs: build
     if:  ${{ !cancelled() && ( needs.build.result == 'success' || needs.build.result == 'skipped' ) && inputs.test_script != '' && inputs.test_image != '' && inputs.run_tests}}
```

Comment:
No change requested.

I am wondering whether we could add another result to the build step, that checks whether we had a 100% sccache hit rate. In that case we can omit running the test step

Reply:
Maybe we can store it in an environment variable